libs/vkd3d/command.c

```diff
@@ -1681,6 +1681,90 @@ static void d3d12_command_list_invalidate_current_pipeline(struct d3d12_command_
     list->current_pipeline = VK_NULL_HANDLE;
 }
 
+enum vkd3d_render_pass_transition_mode
+{
+    VKD3D_RENDER_PASS_TRANSITION_MODE_BEGIN,
+    VKD3D_RENDER_PASS_TRANSITION_MODE_END,
+};
+
+static VkPipelineStageFlags vk_render_pass_barrier_from_view(const struct vkd3d_view *view, const struct d3d12_resource *resource,
+        enum vkd3d_render_pass_transition_mode mode, VkImageMemoryBarrier *vk_barrier)
+{
+    VkPipelineStageFlags stages;
+    VkAccessFlags access;
+
+    if (view->format->vk_aspect_mask & VK_IMAGE_ASPECT_COLOR_BIT)
+    {
+        stages = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+        access = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT | VK_ACCESS_COLOR_ATTACHMENT_READ_BIT;
+    }
+    else
+    {
+        stages = VK_PIPELINE_STAGE_EARLY_FRAGMENT_TESTS_BIT | VK_PIPELINE_STAGE_LATE_FRAGMENT_TESTS_BIT;
+        access = VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_WRITE_BIT | VK_ACCESS_DEPTH_STENCIL_ATTACHMENT_READ_BIT;
```

Comment:
ATTACHMENT_WRITE_BIT is not allowed if using READ_ONLY layout I think?

Reply:
Does it say that anyhwere in the spec? I can't really find anything, but it mentions e.g. that `STORE_OP_STORE` uses `ATTACHMENT_WRITE_BIT`, apparently regardless of the layout.

I can put the `WRITE` bit behind an `if (layout != DEPTH_STENCIL_READ_ONLY_OPTIMAL)` though.

Reply:
If it doesn't fail validation, I suppose it's fine. And yes, STORE_OP_STORE is technically needed, even for read-only resources.